go/chat/utils/utils.go

```diff
@@ -190,47 +190,48 @@ func splitAndNormalizeTLFNameCanonicalize(mctx libkb.MetaContext, name string, p
 	return writerNames, readerNames, extensionSuffix, err
 }
 
-// AttachContactNames retrieves display names for SBS phones/emails that are in the phonebook.
-func AttachContactNames(mctx libkb.MetaContext, participants []chat1.ConversationLocalParticipant) (withContacts []chat1.ConversationLocalParticipant) {
+// AttachContactNames retrieves display names for SBS phones/emails that are in
+// the phonebook. ConversationLocalParticipant structures are modified in place
+// in `participants` passed in argument.
+func AttachContactNames(mctx libkb.MetaContext, participants []chat1.ConversationLocalParticipant) {
+	syncedContacts := mctx.G().SyncedContactList
+	if syncedContacts == nil {
+		mctx.Debug("AttachContactNames: SyncedContactList is nil")
+		return
+	}
 	var contacts []keybase1.ProcessedContact
 	var err error
 	contactsFetched := false
-	for _, participant := range participants {
+	for i, participant := range participants {
 		if isPhoneOrEmail(participant.Username) {
 			if !contactsFetched {
-				contacts, err = mctx.G().SyncedContactList.RetrieveContacts(mctx)
+				contacts, err = syncedContacts.RetrieveContacts(mctx)
 				if err != nil {
-					mctx.Debug("Error fetching contacts: %s", err)
-					return participants
+					mctx.Debug("AttachContactNames: error fetching contacts: %s", err)
+					return
 				}
+				contactsFetched = true
 			}
-			// todo separate phone / email from assertion
-			assertion, err := libkb.ParseAssertionURL(mctx.G().MakeAssertionContext(mctx), participant.Username, true)
-			if err == nil {
-				phoneOrEmail := assertion.GetValue()
-				isPhone := assertion.GetKey() == "phone"
-				contactName := findContactName(contacts, phoneOrEmail, isPhone)
-				participant.ContactName = contactName
-			} else {
-				mctx.Debug("Error parsing assertion: %s", err)
-			}
+			participant.ContactName = findContactName(contacts, participant.Username)
 		}
-		withContacts = append(withContacts, participant)
+		participants[i] = participant
 	}
-	return withContacts
 }
 
-func findContactName(contacts []keybase1.ProcessedContact, phoneOrEmail string, isPhone bool) *string {
+func findContactName(contacts []keybase1.ProcessedContact, assertion string) *string {
+	var result *string
 	for _, contact := range contacts {
```

Comment:
i assume the speed of this isn't an issue but maybe add a comment that this is technically n*m

Reply:
at some point we might want to save `chatHeaders map[assertion]contact` when syncing contacts to simplify this function

Reply:
might have to be map[assertion][]contact?